functions/dialect.go

```diff
@@ -97,6 +97,18 @@ type mapAndSlice[V extensions.FunctionVariant] struct {
 	variantsSlice []V
 }
 
+func nameForInvocationType(invocationType interface{}) string {
```

Comment:
is it better to add this to `localFunctionVariant` interface?

Reply:
Done.  I originally thought about placing it here but didn't like the type methods being mixed with the instance methods.  I gave it a clear name to make the distinction more clear that the method doesn't care about its value.